src/theory/quantifiers/ce_guided_pbe.cpp

```diff
@@ -1399,409 +1693,508 @@ Node CegConjecturePbe::constructBestStringToConcat( std::vector< Node > strs,
+        ret_dt = einfo.getSolved();
+        indent("sygus-pbe-dt", ind);
+        Trace("sygus-pbe-dt") << "return PBE: success : solved "
+                              << d_tds->sygusToBuiltin(ret_dt) << std::endl;
+        Assert(!ret_dt.isNull());
+      }
+      else
+      {
+        // could be conditionally solved
+        std::vector<Node> subsumed_by;
+        einfo.d_term_trie.getSubsumedBy(this, x.d_vals, true, subsumed_by);
+        if (!subsumed_by.empty())
+        {
+          ret_dt = constructBestSolvedTerm(subsumed_by, x);
+          indent("sygus-pbe-dt", ind);
+          Trace("sygus-pbe-dt") << "return PBE: success : conditionally solved"
+                                << d_tds->sygusToBuiltin(ret_dt) << std::endl;
+        }
+        else
+        {
+          indent("sygus-pbe-dt-debug", ind);
+          Trace("sygus-pbe-dt-debug")
+              << "  ...not currently conditionally solved." << std::endl;
+        }
+      }
+    }
+    if (ret_dt.isNull())
+    {
+      if (d_tds->sygusToBuiltinType(e.getType()).isString())
+      {
+        // check if a current value that closes all examples
+        // get the term enumerator for this type
+        bool success = true;
+        std::map<Node, EnumInfo>::iterator itet;
+        std::map<EnumRole, Node>::iterator itnt =
+            tinfo.d_enum.find(enum_concat_term);
         if( itnt != itt->second.d_enum.end() ){
           Node et = itnt->second;
           itet = d_einfo.find( et );
+          Assert(itet != d_einfo.end());
         }else{
           success = false;
         }
-      }
-      if( success ){
-        Assert( itet!=d_einfo.end() );
-
-        // get the current examples
-        std::map< Node, std::vector< Node > >::iterator itx = d_examples_out.find( c );
-        Assert( itx!=d_examples_out.end() );
-        std::vector< CVC4::String > ex_vals;
-        x.getCurrentStrings( this, itx->second, ex_vals );
-        Assert( itn->second.d_enum_vals.size()==itn->second.d_enum_vals_res.size() );
-        
-        // test each example in the term enumerator for the type
-        std::vector< Node > str_solved;
-        for( unsigned i=0; i<itet->second.d_enum_vals.size(); i++ ){
-          if( x.isStringSolved( this, ex_vals, itet->second.d_enum_vals_res[i] ) ){
-            str_solved.push_back( itet->second.d_enum_vals[i] );
+        if (success)
+        {
+          // get the current examples
+          std::map<Node, std::vector<Node> >::iterator itx =
+              d_examples_out.find(c);
+          Assert(itx != d_examples_out.end());
+          std::vector<CVC4::String> ex_vals;
+          x.getCurrentStrings(this, itx->second, ex_vals);
+          Assert(itn->second.d_enum_vals.size()
+                 == itn->second.d_enum_vals_res.size());
+
+          // test each example in the term enumerator for the type
+          std::vector<Node> str_solved;
+          for (unsigned i = 0; i < itet->second.d_enum_vals.size(); i++)
```

Comment:
for-each

Reply:
This needs to be an index, since i is used to index d_enum_vals_res as well.

Reply:
missed it, ok